irksome/nystrom_stepper.py

```diff
@@ -0,0 +1,137 @@
+from .base_time_stepper import StageCoupledTimeStepper
+from .bcs import BCStageData, bc2space
+from .deriv import TimeDerivative
+from .tools import component_replace, replace, vecconst
+from firedrake import TestFunction, as_ufl
+import numpy
+from ufl import zero
+
+
+class NystromTableau:
+    def __init__(self, A, b, c, Abar, bbar, order):
+        self.A = A
+        self.b = b
+        self.c = c
+        self.Abar = Abar
+        self.bbar = bbar
+        self.order = order
+
+    @property
+    def num_stages(self):
+        return len(self.b)
+
+
+def butcher_to_nystrom(butch):
+    A = butch.A
+    b = butch.b
+    return NystromTableau(A, b, butch.c, A @ A, A.T @ b, butch.order)
```

Comment:
Do we want a `partitioned_to_nystrom` that takes 2 BTs and spits out the Nystrom Tableau that results?

Reply:
Partitioned methods are for systems already written in 1st order form, just split into two pieces (and so applicable in principle to second order problems).  If we had partitioned methods, we could (and probably should) do this conversion, and maybe have the stepper automatically call the conversion for us.

Reply:
I am thinking about schemes like Störmer-Verlet and other Lobatto IIIA-IIIB pairs, where there is an underlying partitioned scheme (that we have yet to implement), that leads to a sensible Nyström method.  I'd rather be able to construct those programatically (admittedly we don't have Lobatto IIIB implemented yet) than in an ad-hoc way.  I'm happy to push that constructor down the road, though, until it is more relevant.

Reply:
Agreed these are good and worth doing (if we can figure out efficiency) and once we do, the tableaux can programmatically be munged into Nystrom.  I think we should wait till we tackle those later